src/SfxWeb/src/app/views/cluster/federation/federation.component.ts

```diff
@@ -0,0 +1,656 @@
+import { TimeUtils } from 'src/app/Utils/TimeUtils';
+import { BaseControllerDirective } from 'src/app/ViewModels/BaseController';
+import { Network} from "vis-network/standalone"
+
+// structure that has all the relevant info about particular node
+export interface NodeData {
+  name?: string;
+  id?: string;
+  status?: string;
+  phase?: string;  
+  isSeed?: boolean;
+  openFailed?: string;
+  FM?: boolean;
+  FMM?: boolean;
+  leaseStates?: Map<string, string>;
+}
+
+// all global variables and functions below are in needs of network click listener
+// even it is called inside nonstatic method, it is local function and does not recognize "this"
+// passing pointers to functions also lose information about "this", thus they are here global
+
+// two hashmaps, since one group of events provides node names and other one node ids
+let nodesMap = new Map<string, NodeData>();
+let nameToIdMap = new Map<string, string>();
+let neighSize: number = 2;
+
+// data shown on node hover
+function formTitle(elem: [string, NodeData]): string {
+  return "Id: " + elem[1].id +
+  "\nStatus: " + elem[1].status + 
+  "\nPhase: " + elem[1].phase +
+  (elem[1].FM ? "\nFM" : "") + 
+  (elem[1].FMM ? "\nFMM" : "") +
+  (elem[1].isSeed ? "\nSEED" : "") + 
+  (elem[1].openFailed != "" ? "\nOpenFailed: " + elem[1].openFailed : "");
+}
+
+function getNodeColor(node: NodeData): any {
+  if (node.status == "Up") return {
+    border: "rgba(10,141,16,1)",
+    background: "rgba(124,255,131,1)",
+    highlight: {
+      border: "rgba(10,141,16,1)",
+      background: "rgba(0,255,1,1)",
+    },
+    hover: {
+      border: "rgba(10,141,16,1)",
+      background: "rgba(0,255,1,1)",
+    }
+  };
+  else if (node.status == "Gray") return {
+    border: "rgba(68,71,73,1)",
+    background: "rgba(120,133,142,1)",
+    highlight: {
+      border: "rgba(68,71,73,1)",
+      background: "rgba(116,123,128,1)",
+    },
+    hover: {
+      border: "rgba(68,71,73,1)",
+      background: "rgba(116,123,128,1)",
+    }
+  };
+  else return {
+    border: "rgba(143,4,4,1)",
+    background: "rgba(191,35,35,1)",
+    highlight: {
+      border: "rgba(143,4,4,1)",
+      background: "rgba(255,0,0,1)",
+    },
+    hover: {
+      border: "rgba(143,4,4,1)",
+      background: "rgba(255,0,0,1)",
+    }
+  };
+}
+
+// logics for creating network
+function drawGraph(mode: string) {
+  nodes = nodes.map((node, index, arr) => {
+    let angle: number;
+    if (mode == "full") {
+      // make the nodes form the structure of ring
+      angle = 2 * Math.PI * (index / arr.length + 0.75);
+      node.x = 400 * Math.cos(angle);
+      node.y = 400 * Math.sin(angle);
+    }
+    else {
+      // make something that visually looks like neighborhood
+      angle = 2 * Math.PI / 3 * (index / arr.length + 1.85);
+      node.x = 1000 * Math.cos(angle);
+      node.y = 1000 * Math.sin(angle);
+    }
+    return node;
+  });
+  let container = document.getElementById("mynetwork");
```

Comment:
we shouldn't use document.getElementById in angular. In this case, I think we could use ViewChild() with template reference variables, see: https://stackoverflow.com/questions/39631594/access-template-reference-variables-from-component-class

Reply:
yeah, thanks for this